content/3_Silver/Prefix_Sums.mdx

```diff
@@ -681,6 +692,299 @@ as expected.
+
+Example: Add two to each point in the interval [2, 4]
+
+Our matrix after adding 2 at our increment (start) point (before computation)
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td>2</td>
+			<td>0</td>
+			<td>0</td>
+			<td>0</td>
+		</tr>
+	</table>
+</center>
+
+Our matrix after adding 2 at our increment (start) point (and computing).
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+		</tr>
+	</table>
+</center>
+
+Our matrix after subtracting 2 at our decrement point (and computing).
+<center>
+	<table className='text-center'>
+		<tr>
+			<td>0</td>
+			<td>0</td>
+			<td>2</td>
+			<td>2</td>
+			<td>2</td>
+			<td>0</td>
+		</tr>
+	</table>
+</center>
+
+In this above problem, we are only incrementing by 1, so our $x$ is consistent. As a result, when we encounter a starting point, we can increment by 1, and for an endpoint, decrement by 1. While a prefix sum array is *technically* not needed, it is a good concept to know when generalizing to construct 2d prefix sums.
+
+### Restaurant Customers Solution
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+ 
+using namespace std;
+ 
+#define ar array
+#define vt vector
+#define f first
+#define s second
+#define lb lower_bound
+#define ub upper_bound
+ 
+#define FOR(i, a, b) for(int i = (a); i < (b); i++)
+#define F0R(i, a) for(int i = 0; i < (a); i++)
+#define trav(a, x) for (auto& a : x)
+ 
+typedef pair<int,int> pi;
+typedef vt<pi> vpi;
+ 
+#define pb push_back
+#define mp make_pair
+#define sz(x) (int)x.size()
+#define beg(x) x.begin()
+#define en(x) x.end()
+#define all(x) beg(x), en(x)
+ 
+int N;
+ 
+int main() {
+    cin >> N;
+ 
+    vpi customers;
+ 
+    F0R(i, N) {
+        int a, b; cin >> a >> b;
+        customers.pb(mp(a, 1));
+        customers.pb(mp(b, -1));
+    }
+ 
+    sort(all(customers));
+ 
+    int count = 0;
+    int ret = 0;
+
+	//note that we don't use a prefix array in this problem because the coordinates are large.
+ 
+    trav(a, customers) {
+        count += a.s;
+        ret = max(ret, count);
+    }
+ 
+    cout << ret;
+}
+```
+
+</CPPSection>
+
+</LanguageSection>
+
+### 2-Dimensional
+
+#### Point
+
+We can apply a similar concept to 2d matrices. Since we define $\texttt{prefix}[i][j]$ as the number of points between $(0,0)$ and $(i, j)$, each point (i, j) is equal to the sum of itself and all the points between $(0, i-1)$ and $(0, j-1)$. However, since we are double counting the square $(i-1, j-1)$, we subtract it from our equation. This can be expressed by
+
+$$
+\texttt{prefix}[i][j] = \texttt{prefix}[i][j] = \texttt{prefix}[i][j] + \texttt{prefix}[i-1][j] + \texttt{prefix}[i][j-1] - \texttt{prefix}[i-1][j-1]
+$$
+
+The algorithm goes as follows:
+
+We first increment every point by 1.
```

Comment:
To avioid confusion, I think you should just include some example points and then simulate them